models/intermediate/int_stripe__deduped_subscription_item.sql

```diff
@@ -0,0 +1,20 @@
+with subscription_item as (
+{{ config(enabled=var('stripe__using_subscriptions', True)) }}    
+
+    select * 
+    from {{ ref('stg_stripe__subscription_item') }}
+
+)
+
+/*
+SUBSCRIPTION_ITEM allows for one-to-many relationships between subscriptions and plans, so we need to dedupe to the subscription_id level
+*/
+
+select
+    subscription_id,
+    source_relation,
+    min(current_period_start) as current_period_start,
+    max(current_period_end) as current_period_end
+        
+from subscription_item
+group by subscription_id, source_relation
```

Comment:
When grouping by we use numerics. This way if the list gets too long then we can leverage the `dbt_utils.group_by()` macro. 

There are cases where we will group by the individual field, but those are select cases.
```suggestion
group by 1, 2
```